spy/vm/astframe.py

```diff
@@ -947,6 +947,64 @@ def eval_expr_UnaryOp(self, unop: ast.UnaryOp) -> W_MetaArg:
         w_opimpl = self.vm.call_OP(unop.loc, w_OP, [wam_v])
         return self.eval_opimpl(unop, w_opimpl, [wam_v])
 
+    def _eval_bool(self, wam: W_MetaArg) -> W_MetaArg:
```

Comment:
this function seems unnecessary.
If you call `self.eval_expr(op.left, varname="@if")`, you ALREADY get a value which has been converted to bool because of `varname="@if"`.
So, wam.w_static_T is already `w_bool`, and `w_typeconv` is always None.
The code after `w_typeconv is None` is NEVER executed (try to put an `assert False` and run the tests)

Reply:
oh, maybe this is what solved the problem `1 and 2` which I explain below. If so, probably it was correct to have it. Sorry for making you doing duplicated work